s3torchbenchmarking/utils/download_and_transform_results.py

```diff
@@ -0,0 +1,391 @@
+                "model": cfg["training"]["model"],
+                "max_epochs": cfg["training"]["max_epochs"],
+                "prefix_uri_suffix": prefix_uri_suffix,
+                "batch_size": cfg["dataloader"]["batch_size"],
+                "num_workers": cfg["dataloader"]["num_workers"],
+                "dataloader_kind": cfg["dataloader"]["kind"],
+                "result": {
+                    "elapsed_time": result["elapsed_time"],
+                    "throughput": result["throughput"],
+                    "utilization": result["utilization"],
+                    "cpu_util_mean": result["utilization"]["cpu_util"]["mean"],
+                    "gpu_util_mean": result["utilization"]["gpu_util"]["mean"],
+                    "cpu_mem_mean": result["utilization"]["cpu_mem"]["mean"],
+                    "gpu_mem_mean": result["utilization"]["gpu_mem"]["mean"],
+                },
+            }
+        )
+    return extracted_data
+
+
+# Helper function to count rows for rowspan
+def count_rows(data: Dict[str, Any]) -> int:
+    if "result" in data:
+        return 1
+    return sum(
+        count_rows(sub_data) for sub_data in data.values() if sub_data != "result"
+    )
+
+
+# Helper function to generate merged table cells
+def generate_table_rows(
+    data: Dict[str, Any], indent: int = 0, col_spans: List[int] = None
+) -> str:
+    if col_spans is None:
+        col_spans = [1] * 7
+
+    html = ""
+    if "result" in data:
+        html += "<tr>"
+        for _ in range(indent):
+            html += "<td></td>"
+        for key, value in data["result"].items():
+            html += f"<td>{value}</td>"
+        html += "</tr>"
+    else:
+        for key, value in data.items():
+            if key == "result":
+                continue
+            span_count = count_rows(value)
+            html += f"<tr><td rowspan='{span_count}'><strong>{key}</strong></td></tr>"
+            for sub_key, sub_value in value.items():
+                html += f"<tr><td rowspan='{count_rows(sub_value)}'>{sub_key}</td>"
+                html += generate_table_rows(sub_value, indent + 2, col_spans)
+                html += "</tr>"
+    return html
+
+
+# Function to read all json files from S3
+def load_data_from_s3(
+    extract_fields_function: callable,
+    bucket_name: str,
+    prefix: str,
+) -> ExtractedData:
+    object_keys = get_object_keys(bucket_name, prefix)
+    all_data = []
+
+    for object_key in object_keys:
+        json_data = read_json_from_s3(bucket_name, object_key)
+        all_data.extend(extract_fields_function(json_data))
+    return all_data
+
+
+# Helper function to save results to json
+def save_data_to_simple_json(all_data: ExtractedData, file_name: str) -> None:
+    with open(file_name, "w", encoding="utf-8") as file:
+        json.dump(all_data, file, indent=4)
+
+
+# Function to generate html page that represent output data in table view
+def save_data_to_html(
+    all_data: ExtractedData,
```

Comment:
To make this more readable and separate concerns, should we move the HTML handling logic in a separate class?